### PR TITLE
Fix feature flag doc formatting

### LINKED ISSUE #3971: Feature flag docs lists are a mess
The feature flag docs here: https://knative.dev/docs/serving/feature-flags/ have multiple lists that are broken and instead render as "* list item 1 * list item 2...etc"

I think this is just an issue with spacing that needs to be fixed throughout the page.

### PR DESCRIPTION
**Pull request by abrennan89**
Fixes  #3971

FYI: This PR needs a follow up issue to decide whether to use `-` or `*` for bullet lists in docs @snneji @RichardJJG 
Both are accepted but we should probably just choose one to keep docs consistent; any preference?

### CODE REVIEW

File: docs/serving/feature-flags.md [hunk 1]
```diff
@@ -1,20 +1,18 @@
...
-title: "Feature/Extension Flags"
-weight: 50
-type: "docs"
----
+# Feature and extension flags
 
-# Feature/Extension Flags
-
-Knative is deliberate about the concepts it incorporates into its core API. The API aims to be portable and abstracts away the specificities of each users' implementation. That being said, the Knative API should empower users to surface extra features and extensions possible within their platform of choice.
+The Knative API is designed to be portable, and abstracts away specific implementation details for user deployments, while intending to empower users to surface extra features and extensions that are possible within their platform of choice.
```
Comment (snneji) [inline docs/serving/feature-flags.md:3]: This sentence is pretty long. Consider breaking it up a bit.

File: docs/serving/feature-flags.md [hunk 2]
```diff
@@ -1,20 +1,18 @@
----
-title: "Feature/Extension Flags"
-weight: 50
-type: "docs"
----
+# Feature and extension flags
```
Comment (snneji) [inline docs/serving/feature-flags.md:1]: Maybe add this new title to the `nav.yaml`

**Review by snneji — COMMENTED** [review]
Bullet point formatting looks good. Just a couple of suggestions:

- Add the updated title to the `nav.yaml`
- Break up the long intro sentence

**Comment by abrennan89** [comment]
Great suggestions, thanks @snneji 
Should be good to go now

**Pull request merged**